packages/admin/admin/src/EditDialog.stackTabsAndGrid.test.tsx

```diff
@@ -0,0 +1,199 @@
+import { Add, Edit } from "@comet/admin-icons";
+import { Button, createTheme, IconButton } from "@mui/material";
+import { DataGrid } from "@mui/x-data-grid";
+import { screen } from "@testing-library/react";
+import { createMemoryHistory } from "history";
+import { ReactNode, RefObject, useRef } from "react";
+import { useIntl } from "react-intl";
+import { Router } from "react-router";
+import { render } from "test-utils";
+
+import { MainContent } from "./common/MainContent";
+import { ToolbarActions } from "./common/toolbar/actions/ToolbarActions";
+import { ToolbarAutomaticTitleItem } from "./common/toolbar/automatictitleitem/ToolbarAutomaticTitleItem";
+import { ToolbarBackButton } from "./common/toolbar/backbutton/ToolbarBackButton";
+import { DataGridToolbar } from "./common/toolbar/DataGridToolbar";
+import { ToolbarFillSpace } from "./common/toolbar/fillspace/ToolbarFillSpace";
+import { StackToolbar } from "./common/toolbar/StackToolbar";
+import { EditDialog } from "./EditDialog";
+import { IEditDialogApi } from "./EditDialogApiContext";
+import { FinalForm } from "./FinalForm";
+import { TextField } from "./form/fields/TextField";
+import { MuiThemeProvider } from "./mui/ThemeProvider";
+import { SaveBoundary } from "./saveBoundary/SaveBoundary";
+import { SaveBoundarySaveButton } from "./saveBoundary/SaveBoundarySaveButton";
+import { StackPage } from "./stack/Page";
+import { Stack } from "./stack/Stack";
+import { StackLink } from "./stack/StackLink";
+import { StackSwitch } from "./stack/Switch";
+import { RouterTab, RouterTabs } from "./tabs/RouterTabs";
+
+describe("EditDialog with Stack, Router Tabs and Grid", () => {
+    const history = createMemoryHistory();
+
+    type DialogProps = {
+        dialogApiRef: RefObject<IEditDialogApi>;
+    };
+
+    const AddProductDialog = ({ dialogApiRef }: DialogProps) => {
+        const intl = useIntl();
+
+        return (
+            <EditDialog
+                ref={dialogApiRef}
+                title={intl.formatMessage({ id: "addProductDialog.title", defaultMessage: "Add a new product" })}
+                componentsProps={{ dialog: { scroll: "paper", fullWidth: true } }}
+            >
+                {() => {
+                    return (
+                        <FinalForm
+                            mode="add"
+                            onSubmit={() => {
+                                // Submit logic
+                            }}
+                            onAfterSubmit={() => {
+                                dialogApiRef.current?.closeDialog();
```

Comment:
I don't think is this needed, the dialog auto closes after submit (see https://github.com/vivid-planet/comet/blob/main/packages/admin/admin/src/EditDialog.tsx#L134)

Reply:
Yes you are right, it was copied from a Storybook Story. I will remove it.